### PR TITLE
Improve docs in various ways

### PR DESCRIPTION
**Pull request by pquentin**
Please review commit by commit, I can easily revert any of them.

### CODE REVIEW

File: docs/developing.rst [hunk 1]
```diff
@@ -34,15 +34,6 @@ Rally uses automatic code formatters. They're enforced by ``make lint`` and you
 
 However, consider using editor integrations to do it automatically: you'll need to configure `black <https://black.readthedocs.io/en/stable/integrations/editors.html>`_ and `isort <https://github.com/PyCQA/isort/wiki/isort-Plugins>`_.
 
-Known Issues
```
Comment (b-deam) [inline docs/developing.rst:37]: Is there a specific reason we removed the known issue(s)?

File: docs/metrics.rst [hunk 2]
```diff
@@ -4,7 +4,7 @@ Metrics
 Metrics Records
 ---------------
 
-At the end of a race, Rally stores all metrics records in its metrics store, which is a dedicated Elasticsearch cluster. Rally stores the metrics in the indices ``rally-metrics-*``. It will create a new index for each month.
+At the end of a race, Rally stores all metrics records in its metrics store, which is a dedicated Elasticsearch cluster, ie. not the cluster where Rally ran its benchmarks. Rally stores the metrics in the indices ``rally-metrics-*``. It will create a new index for each month.
```
Comment (b-deam) [inline docs/metrics.rst:7]: Perhaps worth adding a link here to the configuration section on reporting/metrics (https://esrally.readthedocs.io/en/stable/configuration.html#reporting) ?

**Review by b-deam — COMMENTED** [review]
Only a couple of nits/suggestions, otherwise LGTM.

File: docs/developing.rst (hunk 1 above)
Comment (pquentin) [reply docs/developing.rst:37]: I mentioned it in the commit, the issue is fixed in pyenv

Comment (b-deam) [reply docs/developing.rst:37]: Do'h 🤦 . Thanks for confirming :-)

File: docs/metrics.rst (hunk 2 above)
Comment (pquentin) [reply docs/metrics.rst:7]: Good idea, will do

File: docs/developing.rst (hunk 1 above)
Comment (pquentin) [reply docs/developing.rst:37]: Reading that again, I see that my comment is terse and sounds like it was obvious. It was not, and GitHub's UI makes it difficult to see the commit descriptions. Sorry about that!

File: docs/metrics.rst (hunk 2 above)
Comment (pquentin) [reply docs/metrics.rst]: I tried to use Sphinx's internal links, but failed to link to the subsection, so I linked directly to the online docs.

Please take another look!

File: docs/developing.rst (hunk 1 above)
Comment (b-deam) [reply docs/developing.rst:37]: > Reading that again, I see that my comment is terse and sounds like it was obvious. It was not, and GitHub's UI makes it difficult to see the commit descriptions. Sorry about that!

Haha, not a problem. [I did not assume any malice :- `01.02, /FORMAT` ;-).](https://www.elastic.co/about/our-source-code)

File: docs/telemetry.rst [hunk 3]
```diff
@@ -31,6 +31,8 @@ You probably want to gain additional insights from a race. Therefore, we have ad
    shard-stats                 Shard Stats                 Regularly samples nodes stats at shard level
    data-stream-stats           Data Streams Stats          Regularly samples data streams stats
 
+.. warning::
```
Comment (danielmitterdorfer) [inline docs/telemetry.rst:34]: This should be a separate warning block, maybe before the output of the command line example. The sentence below ("Keep in mind that each telemetry device may incur a runtime overhead which can skew results.") is actually part of the command line output of `esrally list telemetry` and should not be removed.

**Review by danielmitterdorfer — COMMENTED** [review]
Thanks! Looks mostly fine, I left a few minor comments.

File: docs/telemetry.rst (hunk 3 above)
Comment (pquentin) [reply docs/telemetry.rst]: The alternative I see is duplicating the warning, please tell me if that's better

Comment (danielmitterdorfer) [reply docs/telemetry.rst]: Yes, I'd prefer a separate warning (above the example) directly in the docs. This fact is important enough that it warrants a warning in the docs and in the command line output IMHO.

Comment (pquentin) [reply docs/telemetry.rst]: Sorry, I meant "I've updated the docs like you said, please take a look". I wasn't trying to ask you to repeat your argument a second time, hoping that would change your mind :D

**Review by danielmitterdorfer — APPROVED** [review]
LGTM

**Pull request merged**